Simulator/simulator/garden.py

```diff
@@ -662,7 +665,7 @@ def save_coverage_and_diversity(self):
         coverage = total_cc / (self.N * self.M)
         prob = cc_per_plant_type[np.nonzero(cc_per_plant_type)] / total_cc
         entropy = np.sum(-prob * np.log(prob))
-        diversity = entropy / np.log(len(self.plant_types))  # normalized entropy
+        diversity = entropy / np.maximum(np.log(len(self.plant_types)), 1)  # normalized entropy
```

Comment:
Why max with 1?  If we have 4 plant types, then ln(4) = 1.38629436.  Let's say we have prob values of 0.2, 0.2, 0.5 and 0.1  Then the resulting total diversity on computed from line 668 will be 1.22 instead of 0.88.

Reply:
Right! That's an error.